File: unity/Assets/Scripts/Multiplayer/Matchmaking/Requests/MatchmakingPermissionResponse.cs
using Michitai.Multiplayer.Errors;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Michitai.Multiplayer.Matchmaking.Requests
{
    /// <summary>
    /// Response returned when a host responds to a join request (approve/reject).
    /// </summary>
    [System.Serializable]
    public class MatchmakingPermissionResponse : ApiResponse<EMatchmakingResponseError>
    {
        /// <summary>
        /// Confirmation message for the response action.
        /// </summary>
        public string message;

        /// <summary>
        /// The ID of the join request that was responded to.
        /// </summary>
        public string request_id;

        /// <summary>
        /// The action taken (Approve or Reject).
        /// </summary>
        public string action;
    }
}

File: java/src/main/java/com/michitai/multiplayer/matchmaking/requests/MatchmakingPermissionResponse.java
package com.michitai.multiplayer.matchmaking.requests;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.michitai.multiplayer.ApiResponse;

/**
 * Response returned when a host responds to a join request (approve/reject).
 */
public class MatchmakingPermissionResponse extends ApiResponse {
    @JsonProperty("message")
    private String message;

    @JsonProperty("request_id")
    private String requestId;

    @JsonProperty("action")
    private String action;

    public String getMessage() {
        return message;
    }

    public void setMessage(String message) {
        this.message = message;
    }

    public String getRequestId() {
        return requestId;
    }

    public void setRequestId(String requestId) {
        this.requestId = requestId;
    }

    public String getAction() {
        return action;
    }

    public void setAction(String action) {
        this.action = action;
    }
}
